packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -137,18 +138,24 @@ export class CacheDataClient implements IDataClient {
   private readonly cacheServiceErrorMapper: CacheServiceErrorMapper;
   private readonly interceptors: Interceptor[];
   private readonly streamingInterceptors: Interceptor[];
+  private semaphore: Semaphore;
 
   /**
    * @param {CacheClientProps} props
    * @param dataClientID
    */
-  constructor(props: CacheClientPropsWithConfig, dataClientID: string) {
+  constructor(
+    props: CacheClientPropsWithConfig,
+    dataClientID: string,
+    semaphore: Semaphore
+  ) {
     this.configuration = props.configuration;
     this.credentialProvider = props.credentialProvider;
     this.logger = this.configuration.getLoggerFactory().getLogger(this);
     this.cacheServiceErrorMapper = new CacheServiceErrorMapper(
       props.configuration.getThrowOnErrors()
     );
+    this.semaphore = semaphore;
```

Comment:
Same for this name, or at least document that it's for rate limiting in the field's docstring.